### PR TITLE
Add support for authentication on Quakenet

### PR DESCRIPTION
**Pull request by Meicceli**
I added a couple of rows of simple code which add the support for authenticating your Sopel on Quakenet.

Modified files are `core_tasks.py`, and `core_section.py`. The modifications to the code should be simple enough to understand just by looking at them.

Works on my machine™ (`Sopel v. 6.3.1`, installed via pip).

### CODE REVIEW

**Comment by maxpowa** [comment]
Judging from https://www.quakenet.org/help/q-commands/auth, Quakenet appears to prefer `AUTH <user> <pass>` over `PRIVMSG`ing Q directly.

**Comment by Meicceli** [comment]
Welp, let's do that instead. I'll just finish my meat pie and this episode of always sunny first.

**Comment by Meicceli** [comment]
So that's done. Works well on my Sopel 6.3.1 (pip). The new code's now almost identical to the AuthServ bit lol.

File: sopel/config/core_section.py [hunk 1]
```diff
@@ -50,10 +50,10 @@ class CoreSection(StaticSection):
     capabilities."""
 
     auth_method = ChoiceAttribute('auth_method', choices=[
-        'nickserv', 'authserv', 'sasl', 'server'])
+        'nickserv', 'authserv', 'Q','sasl', 'server'])
```
Comment (maxpowa) [inline sopel/config/core_section.py]: There's a space missing after the comma here, just minor code style issue.

**Comment by Meicceli** [comment]
Fix'd

**Comment by maxpowa** [comment]
Thanks.

**Pull request merged**